ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/Party.kt

```diff
@@ -0,0 +1,29 @@
+package net.corda.v5.ledger.consensual
+
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.base.types.MemberX500Name
+import net.corda.v5.crypto.toStringShort
+import java.security.PublicKey
+
+/**
+ * Representation of a Party.
+ */
+@DoNotImplement
+@CordaSerializable
+interface Party {
+    /**
+     * @property name X500 Name of the party.
+     */
+    val name: MemberX500Name
+
+    /**
+     * @property owningKey The [PublicKey] owned by the party.
+     */
+    val owningKey: PublicKey
+
+    /**
+     * Concise, readable String representation of a Party.
+     */
+    val description: String get() = "$name (owningKey = ${owningKey.toStringShort()})"
```

Comment:
Are we better off just overriding `toString` on the implementation?

Reply:
I think I got it from here:
https://github.com/corda/corda5/blob/release/os/5.0/application/src/main/kotlin/net/corda/v5/application/identity/Party.kt#L37

Original `toString()` (https://github.com/corda/corda-runtime-os/pull/1715/files#diff-c174eb8fa66f7c7ee5d09ba7b732751c4ec66ff6e802b892cb5000bcbf7f8c75L20)
was just like this in the previous PartyImpl:
`override fun toString() = name.toString()`

Although compatibility/similarity with C4/DP1 is not a big issue in Consensual world.

Anyway I do not have strong feelings.
Should I change it?

Reply:
I have changed it.